ravenframework/Simulation.py

```diff
@@ -771,37 +820,41 @@ def initiateStep(self, stepName):
 
   def executeStep(self, stepInputDict, stepInstance):
     """
-      This method is aimed to execute and finalize the step
-      (it can be splitted in 2 in case needed)
+      This method executes and finalizes the step
       @ In, stepInputDict, dict, the step dictionary
       @ In, stepInstance, instance, step instance
       @ Out, None
     """
     stepInstance.takeAstep(stepInputDict)
-    #---------------here what is going on? Please add comments-----------------
+    # FilePrint has a finalize method that should be run at the end of the step,
+    # run this method for each instance
     for output in stepInputDict['Output']:
-      if "finalize" in dir(output):
-        output.finalize()
-    self.raiseAMessage('-'*2+' End step {0:50} '.format(stepInstance.name+' of type: '+stepInstance.type)+2*'-'+'\n')#,color='green')
+      if "_printer" in output.__dict__:
+        output._printer.finalize()
+
+    self.raiseAMessage(f'-- End step {stepInstance.name} of type: {stepInstance.type} --\n')
 
   def finalizeSimulation(self):
     """
       This method is called at end of the simulation to finalize it
-      It is in charge of shutting dow the job handler and clean up the execution
+      It is in charge of shutting down the job handler and cleaning up the execution
       @ In, None
       @ Out, None
     """
     self.jobHandler.shutdown()
     self.messageHandler.printWarnings()
-    # implicitly, the job finished successfully if we got here.
+    # implicitly, the job finished successfully if we got here -- self.pollingThread.is_Alive()
+    # returns False and no new jobs can be queued.
     self.writeStatusFile()
+    self.ranPreviously = True
 
   def stepSequence(self):
     """
       Return step sequence
       @ In, None
       @ Out, stepSequence, list(str), list of step in sequence
     """
+
     return self.__stepSequenceList
```

Comment:
I like all of my return statements to be offset with whitespace. I find it more readable. Since I'm making edits to `Simulation.py` I'll get rid of it.